h2o-algos/src/main/java/hex/tree/isofor/IsolationForest.java

```diff
@@ -71,7 +74,27 @@
     }
     _parms._distribution = DistributionFamily.gaussian;
     if (_parms._contamination != -1 && (_parms._contamination <= 0 || _parms._contamination > 0.5)) {
-      error("_contamination", "Contamination parameter needs to be in range (0, 0.5] or undefined (-1); but it is " + _parms._contamination); 
+      error("_contamination", "Contamination parameter needs to be in range (0, 0.5] or undefined (-1); but it is " + _parms._contamination);
+    }
+    if (_parms._valid != null) {
+      if (_parms._response_column == null) {
+        error("_response_column", "Response column needs to be defined when using a validation frame.");
+      } else if (expensive && vresponse() == null) {
+        error("_response_column", "Validation frame is missing response column `" + _parms._response_column + "`.");
+      }
+      if (_parms._contamination > 0) {
+        error("_contamination", "Contamination parameter cannot be used together with a validation frame.");
+      }
+    }
+    if (expensive) {
+      if (vresponse() != null) {
+        if (!vresponse().isBinary() || vresponse().domain()==null) {
+          error("_response_column", "The response column of the validation frame needs to have a binary categorical domain (not anomaly/anomaly).");
+        }
+      }
+      if (response() != null) {
+        error("_training_frame", "Training frame should not have a response column");
```

Comment:
This leads me back to my comment in `IsolationForestV3` about the parameter naming. Naming it differently, e.g. `validation_response_column` would send out a clear message the parameter has nothing to do with training phase.

Also, if the training data contains the column, shouldn't we just ignore it ? Just a question we should have answered before commiting to this. It's probably better to fail and let the user know, so I vote for keeping the error in place.